tests/Linting/Linters/ModelMethodOrderTest.php

```diff
@@ -11,46 +11,104 @@ class ModelMethodOrderTest extends TestCase
     /** @test */
     function catches_wrong_order_for_model_methods()
     {
-        $file = <<<file
+        $file = <<<PHP
 <?php
 
 namespace App;
 
 use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\HasOne;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Builder;
 
 class Thing extends Model
 {
+    public static function booted()
+    {
+
+    }
+
+    public static function booting()
+    {
+
+    }
+
     public static function boot()
     {
-     
+
     }
-    
+
+    public function scopeWhereInactive(Builder \$query)
+    {
+        return \$query->where('is_active', false);
+    }
+
     public function setFirstNameAttribute(\$value)
     {
         \$this->attributes['first_name'] = strtolower(\$value);
     }
-    
+
+    public function category(): BelongsTo
+    {
+        return \$this->belongsTo('App\Category');
+    }
+
+    public function scopeWhereActive(Builder \$query)
+    {
+        return \$query->where('is_active', true);
+    }
+
     public function setLastNameAttribute(\$value)
     {
         \$this->attributes['first_name'] = strtolower(\$value);
     }
-    
-    public function getFirstNameAttribute(\$value)
+
+    public function activate()
     {
-        return ucfirst(\$value);
+        \$this->is_active = true;
     }
-    
-    public function phone()
+
+    public function phone(): HasOne
     {
         return \$this->hasOne('App\Phone');
     }
+
+    public function tags(): HasMany
+    {
+        return \$this->hasMany('App\Tag');
```

Comment:
Could we potentially match on the common relationship method names (e.g. `hasMany` etc) as a best effort match strategy?

Reply:
How about both!?
But yeah, will try to make this work. Have dig into the parser lib. 😅

Reply:
Oh definitely both if possible haha, but yea it is a bit trickier 😅

Reply:
https://github.com/tightenco/tlint/pull/163/commits/3b1940a01eab268bfae14d38f93e405ad292f666

Have also made the `boot` ones stricter and added a `custom` test that flags every `abstract`, `private`, `protected` or `static` method as `custom`. This way the other tests don't have to test for method visibility or anything.
The `$this->tests` declaration is also commented that the order is important and comments also every of the 3 blocks. Have also added some more methods to the test that should be ordered the right way.